From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: a821 <a821@nospam.mail.de>
Date: Fri, 26 Sep 2025 20:07:30 +0200
Subject: [PATCH] fix many deprecation errors

* remove deprecated `bdist_wheel`
* remove deprecated `distutils` imports
* remove deprecated `TestCommand`
* remove pinned build dependencies (unneeded for arch)
---
 pyproject.toml |  5 ++---
 setup.cfg      |  3 ---
 setup.py       | 21 ++-------------------
 3 files changed, 4 insertions(+), 25 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index 9e5511c..ca479fc 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,8 +1,7 @@
 [build-system]
 requires = [
-    "setuptools; python_version != '3.3'",
-    "setuptools<40.0; python_version == '3.3'",
-    "setuptools_scm<8.0"
+    "setuptools",
+    "setuptools_scm"
 ]
 build-backend = "setuptools.build_meta"
 
diff --git a/setup.cfg b/setup.cfg
index df1c8d2..5f932ec 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -1,6 +1,3 @@
-[bdist_wheel]
-universal = 1
-
 [metadata]
 name = python-dateutil
 description = Extensions to the standard Python datetime module
diff --git a/setup.py b/setup.py
index 069514f..d95b2e4 100644
--- a/setup.py
+++ b/setup.py
@@ -4,9 +4,7 @@ import os
 
 import setuptools
 from setuptools import setup, find_packages
-from setuptools.command.test import test as TestCommand
 
-from distutils.version import LooseVersion
 import warnings
 
 import io
@@ -15,18 +13,6 @@ import sys
 if isfile("MANIFEST"):
     os.unlink("MANIFEST")
 
-if LooseVersion(setuptools.__version__) <= LooseVersion("24.3"):
-    warnings.warn("python_requires requires setuptools version > 24.3",
-                  UserWarning)
-
-
-class Unsupported(TestCommand):
-    def run(self):
-        sys.stderr.write("Running 'test' with setup.py is not supported. "
-                         "Use 'pytest' or 'tox' to run the tests.\n")
-        sys.exit(1)
-
-
 ###
 # Load metadata
 
@@ -51,8 +37,5 @@ setup(
           'write_to': 'src/dateutil/_version.py',
       },
       ## Needed since doctest not supported by PyPA.
-      long_description = README,
-      cmdclass={
-          "test": Unsupported
-      }
-      )
+      long_description = README
+)
-- 
2.51.0

